qiskit/assembler/disassemble.py

```diff
@@ -59,6 +61,31 @@ def _disassemble_circuit(qobj) -> CircuitModule:
     return CircuitModule((_experiments_to_circuits(qobj), run_config, user_qobj_header))
 
 
+def _qobj_to_circuit_cals(qobj, pulse_lib, param_pulses):
+    """Return circuit calibrations dictionary from qobj/exp config calibrations."""
+    qobj_cals = qobj.config.calibrations.to_dict()["gates"]
+    converter = QobjToInstructionConverter(pulse_lib)
+
+    qc_cals = {}
+    for gate in qobj_cals:
+        config = (tuple(gate["qubits"]), tuple(gate["params"]))
+        cal = {
+            config: pulse.Schedule(
+                name="%s %s %s" % (gate["name"], str(gate["params"]), str(gate["qubits"]))
+            )
+        }
+        for instruction in gate["instructions"]:
+            schedule = (
+                converter.convert_parametric(PulseQobjInstruction.from_dict(instruction))
+                if "pulse_shape" in instruction and instruction["pulse_shape"] in param_pulses
+                else converter(PulseQobjInstruction.from_dict(instruction))
+            )
+            cal[config] = cal[config].insert(schedule.ch_start_time(), schedule)
+        qc_cals.setdefault(gate["name"], cal)
```

Comment:
I think this may be a bug. If a gate has a key already, (e.g. `{'rx': {((0,), ()): Schedule(...)}}`), then the value of that gate will be returned and `cal` will be ignored.
```suggestion
        if gate["name"] in qc_cals:
            qc_cals[gate["name"]].update(cal)
        else:
            qc_cals[gate["name"]] = cal
```

Reply:
![Screen Shot 2021-05-24 at 11 48 21 AM](https://user-images.githubusercontent.com/25019973/119372998-f1e63400-bc85-11eb-9ed4-1039615c6d2e.png)

ah i think this is why i originally tacked on the `.update(cal)` to the setdefault. this should do the trick then,r ight?

Reply:
Let's avoid `setdefault` because it's a method that is meant to grab a value, like the `get` method, but you are trying to add a value, not get one.

Reply:
makes sense!